### PR TITLE
Tr/minor tempest regridder changes

### PR DESCRIPTION
**Pull request by imreddyTeja**
## Purpose 

- ~~Make TempestRegridder support multiple vars (maybe)~~ This is likely not needed. It should be its own PR if we ever decide it is needed, and the implementation/interface requires planning to reflect the design of InterpolatonsRegridder
- ~~Make outfile_root a kwarg for TempestRegridder~~ Just remove it.
- ~~improve on read_available_dates~~
- Add tests for TempestRegridder and read_available_dates
- Replace `yyyymmdd_to_datetime` with Dates built in parsing

## Content

----
- [x] I have read and checked the items on the review checklist.

### CODE REVIEW

File: ext/TempestRegridderExt.jl [hunk 1]
```diff
@@ -72,10 +72,10 @@ function Regridders.TempestRegridder(
     input_file::AbstractString;
     regrid_dir::AbstractString,
     mono::Bool = true,
+    outfile_root::AbstractString = "cgll",
```
Comment (Sbozzolo) [inline ext/TempestRegridderExt.jl:75]: Why is this needed?

Comment (imreddyTeja) [reply ext/TempestRegridderExt.jl:75]: There is no real need for this to be a kwarg, or `outfile_root` to be changed in general. 

Files saved during Regridding currently have a name structure as 

```
hd_outfile_root * "_" * varname * "_" * string(time) * ".hdf5"
```

but `hd_outfile_root` is automatically set to `varname`, so the path becomes

```
varname * "_" * varname * "_" * string(time) * ".hdf5"
```
, which seems odd to me. The changes/potential changes in this PR came up in [this](https://github.com/CliMA/ClimaCoupler.jl/pull/1109) PR

Comment (Sbozzolo) [reply ext/TempestRegridderExt.jl:75]: Ah okay! I guess we can directly removed the `hd_outfile_root` then

Comment (juliasloan25) [reply ext/TempestRegridderExt.jl:75]: I agree it's okay to remove it. It would be good to have an organized naming of regridded files when/if we support regridding a file that contains multiple variables. But that can be accomplished using `regrid_dir` rather than `outfile_root`.

**Review by juliasloan25 — APPROVED** [review]

**Review by Sbozzolo — APPROVED** [review]
Thank you!
